ui/job-view/redux/stores/notifications.js

```diff
@@ -0,0 +1,109 @@
+import {
+  clearNotification,
+  clearExpiredTransientNotifications,
+} from '../../../helpers/notifications';
+
+const MAX_STORED_NOTIFICATIONS = 40;
+const LOCAL_STORAGE_KEY = 'notifications';
+
+const types = {
+  NOTIFY: 'NOTIFY',
+  CLEAR: 'CLEAR',
+  CLEAR_EXPIRED_TRANSIENTS: 'CLEAR_EXPIRED_TRANSIENTS',
+  CLEAR_ALL_ON_SCREEN: 'CLEAR_ALL_ON_SCREEN',
+  CLEAR_STORED: 'CLEAR_STORED',
+};
+
+export const notificationActions = {
+  notify: (message, severity, options) => ({
+    type: types.NOTIFY,
+    payload: { message, severity, options },
+  }),
+  clear: index => ({
+    type: types.CLEAR,
+    payload: { index },
+  }),
+  clearExpiredTransients: () => ({
+    type: types.CLEAR_EXPIRED_TRANSIENTS,
+    payload: {},
+  }),
+  clearAllOnScreenNotifications: () => ({
+    type: types.CLEAR_ALL_ON_SCREEN,
+    payload: {},
+  }),
+  clearStored: () => ({
+    type: types.CLEAR_STORED,
+    payload: {},
+  }),
+};
+
+const notify = (state, payload) => {
+  const { message, severity } = payload;
+  const opts = payload.options || {};
+  const { notifications, storedNotifications } = state;
+  const notification = {
+    ...opts,
+    message,
+    severity: severity || 'info',
+    created: Date.now(),
+  };
+  const newNotifications = [notification, ...notifications];
+
+  storedNotifications.unshift(notification);
+  storedNotifications.splice(MAX_STORED_NOTIFICATIONS);
+  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(storedNotifications));
```

Comment:
This might not be relevant here, but for certain things you might not need localStorage since the redux state is held at the top level of the app - if I recall correctly - (for example, when navigating back to a previous view, the last data fetched will still be retained unless there is a deliberate need to fetch data again or that last state is otherwise overwritten).

Reply:
Part of the use of local storage is that it'll persist across multiple days and multiple browser windows and all.  At least that's what some folks had said.  :)  So I think I'll keep it in ``localStorage`` for now, unless you have strong objections.  :)

Reply:
No objection :)